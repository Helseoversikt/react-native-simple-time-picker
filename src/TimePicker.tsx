import * as React from 'react';
import {View, StyleSheet} from 'react-native';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import {PickerProps} from '@react-native-picker/picker/typings/Picker';
import {Picker} from '@react-native-picker/picker';
import {zeroPad} from './utils/zeroPad';

const MAX_HOURS = 23;
const MAX_MINUTES = 59;
const MAX_SECONDS = 59;

export type ValueMap = {
  hours: number;
  minutes: number;
  seconds: number;
};

export interface TimePickerProps extends PickerProps {
  value?: Partial<ValueMap>;
  onChange?: ({hours, minutes, seconds}: ValueMap) => void;
  hoursUnit?: string;
  minutesUnit?: string;
  secondsUnit?: string;
  zeroPadding?: boolean;
  textColor?: string;
  hoursInterval?: number;
  minutesInterval?: number;
  secondsInterval?: number;
  pickerShows?: Array<'hours' | 'minutes' | 'seconds'>;
}

export function TimePicker({
  value,
  onChange,
  hoursUnit,
  minutesUnit,
  secondsUnit,
  zeroPadding = false,
  textColor,
  hoursInterval = 1,
  minutesInterval = 1,
  secondsInterval = 1,
  pickerShows = ['hours', 'minutes'],
  ...others
}: TimePickerProps) {
  if (
    hoursInterval > MAX_HOURS ||
    minutesInterval > MAX_MINUTES ||
    secondsInterval > MAX_SECONDS
  ) {
    throw new Error(
      'value of hoursInterval, minutesInterval or secondsInterval is invalid.',
    );
  }

  const [internalHours, setInternalHours] = React.useState(value?.hours ?? 0);
  const [internalMinutes, setInternalMinutes] = React.useState(
    value?.minutes ?? 0,
  );
  const [internalSeconds, setInternalSeconds] = React.useState(
    value?.minutes ?? 0,
  );

  React.useEffect(() => {
    setInternalHours(value?.hours ?? 0);
    setInternalMinutes(value?.minutes ?? 0);
    setInternalSeconds(value?.seconds ?? 0);
  }, [value]);

  const getLabel = (i: number, unit?: string) => {
    const numString = zeroPadding ? zeroPad(i) : i.toString();
    return `${numString} ${unit ?? ''}`;
  };

  const getHoursItems = () => {
    const items: React.ReactElement[] = [];
    if (!pickerShows.includes('hours')) {
      return items;
    }

    for (let i = 0; i <= MAX_HOURS; i += hoursInterval) {
      items.push(
        <Picker.Item
          testID="hoursItem"
          key={i}
          value={i}
          label={getLabel(i, hoursUnit)}
          color={textColor}
        />,
      );
    }
    return items;
  };

  const getMinutesItems = () => {
    const items: React.ReactElement[] = [];
    if (!pickerShows.includes('minutes')) {
      return items;
    }

    for (let i = 0; i <= MAX_MINUTES; i += minutesInterval) {
      items.push(
        <Picker.Item
          testID="minutesItem"
          key={i}
          value={i}
          label={getLabel(i, minutesUnit)}
          color={textColor}
        />,
      );
    }
    return items;
  };

  const getSecondsItems = () => {
    const items: React.ReactElement[] = [];
    if (!pickerShows.includes('seconds')) {
      return items;
    }

    for (let i = 0; i <= MAX_SECONDS; i += secondsInterval) {
      items.push(
        <Picker.Item
          testID="secondsItem"
          key={i}
          value={i}
          label={getLabel(i, secondsUnit)}
          color={textColor}
        />,
      );
    }
    return items;
  };

  const handleChangeHours = (hours) => {
    setInternalHours(hours);
    const newValue = {
      hours,
      minutes: internalMinutes,
      seconds: internalSeconds,
    };
    onChange?.(newValue);
  };

  const handleChangeMinutes = (minutes) => {
    setInternalMinutes(minutes);
    const newValue = {
      hours: internalHours,
      minutes,
      seconds: internalSeconds,
    };
    onChange?.(newValue);
  };

  const handleChangeSeconds = (seconds) => {
    setInternalSeconds(seconds);
    const newValue = {
      hours: internalHours,
      minutes: internalMinutes,
      seconds,
    };
    onChange?.(newValue);
  };

  return (
    <View style={styles.container}>
      {pickerShows.includes('hours') && (
        <Picker
          testID="hoursPicker"
          style={styles.picker}
          selectedValue={internalHours}
          onValueChange={(itemValue) => handleChangeHours(itemValue)}
          {...others}>
          {getHoursItems()}
        </Picker>
      )}

      {pickerShows.includes('minutes') && (
        <Picker
          testID="minutesPicker"
          style={styles.picker}
          selectedValue={internalMinutes}
          onValueChange={(itemValue) => handleChangeMinutes(itemValue)}
          {...others}>
          {getMinutesItems()}
        </Picker>
      )}

      {pickerShows.includes('seconds') && (
        <Picker
          testID="secondsPicker"
          style={styles.picker}
          selectedValue={internalSeconds}
          onValueChange={(itemValue) => handleChangeSeconds(itemValue)}
          {...others}>
          {getSecondsItems()}
        </Picker>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  picker: {
    flex: 1,
  },
});
