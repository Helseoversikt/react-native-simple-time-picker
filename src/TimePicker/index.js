"use strict";
exports.__esModule = true;
exports.TimePicker = void 0;
var react_1 = require("react");
var react_native_1 = require("react-native");
var picker_1 = require("@react-native-community/picker");
var MAX_HOURS = 24;
var MAX_MINUTES = 60;
function TimePicker(_a) {
    var _b, _c;
    var value = _a.value, onChange = _a.onChange, hoursUnit = _a.hoursUnit, minutesUnit = _a.minutesUnit;
    var _d = react_1.useState((_b = value === null || value === void 0 ? void 0 : value.hours) !== null && _b !== void 0 ? _b : 0), internalHours = _d[0], setInternalHours = _d[1];
    var _e = react_1.useState((_c = value === null || value === void 0 ? void 0 : value.minutes) !== null && _c !== void 0 ? _c : 0), internalMinutes = _e[0], setInternalMinutes = _e[1];
    var getHoursItems = function () {
        var items = [];
        for (var i = 0; i <= MAX_HOURS; i++) {
            items.push(<picker_1.Picker.Item key={i} value={i} label={"" + i + hoursUnit}/>);
        }
        return items;
    };
    var getMinutesItems = function () {
        var items = [];
        for (var i = 0; i <= MAX_MINUTES; i++) {
            items.push(<picker_1.Picker.Item key={i} value={i} label={"" + i + minutesUnit}/>);
        }
        return items;
    };
    var handleChangeHours = function (hours) {
        setInternalHours(hours);
        var newValue = {
            minutes: internalMinutes,
            hours: hours
        };
        onChange === null || onChange === void 0 ? void 0 : onChange(newValue);
    };
    var handleChangeMinutes = function (minutes) {
        setInternalMinutes(minutes);
        var newValue = {
            minutes: minutes,
            hours: internalMinutes
        };
        onChange === null || onChange === void 0 ? void 0 : onChange(newValue);
    };
    return (<react_native_1.View style={styles.container}>
      <picker_1.Picker style={styles.picker} selectedValue={internalHours} onValueChange={function (itemValue) { return handleChangeHours(itemValue); }}>
        {getHoursItems()}
      </picker_1.Picker>
      <picker_1.Picker style={styles.picker} selectedValue={internalMinutes} onValueChange={function (itemValue) { return handleChangeMinutes(itemValue); }}>
        {getMinutesItems()}
      </picker_1.Picker>
    </react_native_1.View>);
}
exports.TimePicker = TimePicker;
var styles = react_native_1.StyleSheet.create({
    container: {
        flexDirection: "row"
    },
    picker: {
        flex: 1
    }
});
