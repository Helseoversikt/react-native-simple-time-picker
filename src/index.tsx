import * as React from 'react';
import {StyleSheet, View} from 'react-native';
import {Picker} from '@react-native-picker/picker';
import {zeroPad} from './utils/zeroPad';

const MAX_HOURS = 23;
const MAX_MINUTES = 59;

export type ValueMap = {
  hours: number;
  minutes: number;
};

export type TimePickerProps = {
  value?: ValueMap;
  onChange?: ({hours, minutes}: ValueMap) => void;
  hoursUnit?: string;
  minutesUnit?: string;
  zeroPadding?: boolean;
  textColor?: string;
};

export function TimePicker({
  value,
  onChange,
  hoursUnit,
  minutesUnit,
  zeroPadding = false,
  textColor,
}: TimePickerProps) {
  const [internalHours, setInternalHours] = React.useState(value?.hours ?? 0);
  const [internalMinutes, setInternalMinutes] = React.useState(
    value?.minutes ?? 0,
  );

  React.useEffect(() => {
    setInternalHours(value?.hours ?? 0);
    setInternalMinutes(value?.minutes ?? 0);
  }, [value]);

  const getLabel = (i, unit) => {
    const numString = zeroPadding ? zeroPad(i) : i.toString();
    return `${numString} ${unit ?? ''}`;
  };

  const getHoursItems = () => {
    const items: React.ReactElement[] = [];
    for (let i = 0; i <= MAX_HOURS; i++) {
      items.push(
        <Picker.Item
          key={i}
          value={i}
          label={getLabel(i, hoursUnit)}
          color={textColor}
        />,
      );
    }
    return items;
  };

  const getMinutesItems = () => {
    const items: React.ReactElement[] = [];
    for (let i = 0; i <= MAX_MINUTES; i++) {
      items.push(
        <Picker.Item
          key={i}
          value={i}
          label={getLabel(i, minutesUnit)}
          color={textColor}
        />,
      );
    }
    return items;
  };

  const handleChangeHours = (hours) => {
    setInternalHours(hours);
    const newValue = {
      minutes: internalMinutes,
      hours,
    };
    onChange?.(newValue);
  };

  const handleChangeMinutes = (minutes) => {
    setInternalMinutes(minutes);
    const newValue = {
      minutes,
      hours: internalHours,
    };
    onChange?.(newValue);
  };

  return (
    <View style={styles.container}>
      <Picker
        style={styles.picker}
        selectedValue={internalHours}
        onValueChange={(itemValue) => handleChangeHours(itemValue)}>
        {getHoursItems()}
      </Picker>
      <Picker
        style={styles.picker}
        selectedValue={internalMinutes}
        onValueChange={(itemValue) => handleChangeMinutes(itemValue)}>
        {getMinutesItems()}
      </Picker>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  picker: {
    flex: 1,
  },
});
