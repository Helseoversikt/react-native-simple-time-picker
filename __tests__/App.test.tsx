import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {TimePicker} from '../src';

describe('ReactNativeSimpleTimePicker', () => {
  it('should render', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker pickerShows={['hours', 'minutes', 'seconds']} />,
    );
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render all hours and minutes items', () => {
    const wrapper = shallow<typeof TimePicker>(<TimePicker />);
    expect(wrapper.find({testID: 'hoursItem'}).length).toBe(24);
    expect(wrapper.find({testID: 'minutesItem'}).length).toBe(60);
    expect(wrapper.find({testID: 'secondsItem'}).length).toBe(0);
  });

  it('should render pickers', () => {
    const wrapper = shallow<typeof TimePicker>(<TimePicker />);
    expect(wrapper.find({testID: 'hoursPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'minutesPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'secondsPicker'}).isEmptyRender()).toBe(true);
    expect(wrapper.find({testID: 'ampmPicker'}).isEmptyRender()).toBe(true);
  });

  it('should render pickers with pickerShows prop', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker pickerShows={['hours', 'minutes', 'seconds']} />,
    );
    expect(wrapper.find({testID: 'hoursPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'minutesPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'secondsPicker'}).isEmptyRender()).toBe(false);
  });

  it('should render pickers with pickerProps e.g. enabled', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker
        pickerShows={['hours', 'minutes', 'seconds']}
        enabled={false}
      />,
    );
    expect(wrapper.find({testID: 'hoursPicker'}).props().enabled).toBe(false);
    expect(wrapper.find({testID: 'minutesPicker'}).props().enabled).toBe(false);
    expect(wrapper.find({testID: 'secondsPicker'}).props().enabled).toBe(false);
  });

  it('should render empty slot with emptyLabel empty string', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker
        pickerShows={['hours', 'minutes', 'seconds']}
        emptyLabel=""
      />,
    );
    expect(wrapper.find({testID: 'hoursItem'}).length).toBe(25);
    expect(wrapper.find({testID: 'hoursItem'}).first().props().label).toBe('');
    expect(wrapper.find({testID: 'minutesItem'}).length).toBe(61);
    expect(wrapper.find({testID: 'minutesItem'}).first().props().label).toBe(
      '',
    );
    expect(wrapper.find({testID: 'secondsItem'}).length).toBe(61);
    expect(wrapper.find({testID: 'secondsItem'}).first().props().label).toBe(
      '',
    );
  });

  it('should not render empty slot with emptyLabel undefined', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker
        pickerShows={['hours', 'minutes', 'seconds']}
        emptyLabel={undefined}
      />,
    );
    expect(wrapper.find({testID: 'hoursItem'}).length).toBe(24);
    expect(wrapper.find({testID: 'hoursItem'}).first().props().label).toBe(
      '0 ',
    );
    expect(wrapper.find({testID: 'minutesItem'}).length).toBe(60);
    expect(wrapper.find({testID: 'minutesItem'}).first().props().label).toBe(
      '0 ',
    );
    expect(wrapper.find({testID: 'secondsItem'}).length).toBe(60);
    expect(wrapper.find({testID: 'secondsItem'}).first().props().label).toBe(
      '0 ',
    );
  });

  it('should render all hours, minutes and seconds items with interval props', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker
        pickerShows={['hours', 'minutes', 'seconds']}
        hoursInterval={5}
        minutesInterval={15}
        secondsInterval={2}
      />,
    );
    expect(wrapper.find({testID: 'hoursItem'}).length).toBe(5);
    expect(wrapper.find({testID: 'minutesItem'}).length).toBe(4);
    expect(wrapper.find({testID: 'secondsItem'}).length).toBe(30);
  });

  it('should fail to render with invalid interval prop', () => {
    expect(() =>
      shallow<typeof TimePicker>(<TimePicker hoursInterval={24} />),
    ).toThrowError();

    expect(() =>
      shallow<typeof TimePicker>(<TimePicker minutesInterval={60} />),
    ).toThrowError();

    expect(() =>
      shallow<typeof TimePicker>(<TimePicker secondsInterval={60} />),
    ).toThrowError();
  });

  it('should render label with units', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker
        hoursUnit="h"
        minutesUnit="m"
        secondsUnit="s"
        pickerShows={['hours', 'minutes', 'seconds']}
      />,
    );

    expect(wrapper.find({testID: 'hoursItem'}).first().props().label).toBe(
      '0 h',
    );
    expect(wrapper.find({testID: 'minutesItem'}).first().props().label).toBe(
      '0 m',
    );
    expect(wrapper.find({testID: 'secondsItem'}).first().props().label).toBe(
      '0 s',
    );
  });

  it('should update state', async () => {
    const wrapper = shallow<typeof TimePicker>(<TimePicker />);

    wrapper.find({testID: 'hoursPicker'}).props().onValueChange(20, 20);
    expect(wrapper.find({testID: 'hoursPicker'}).props().selectedValue).toBe(
      20,
    );

    wrapper.find({testID: 'minutesPicker'}).props().onValueChange(30, 30);
    expect(wrapper.find({testID: 'minutesPicker'}).props().selectedValue).toBe(
      30,
    );

    wrapper.find({testID: 'minutesPicker'}).props().onValueChange(5, 5);
    expect(wrapper.find({testID: 'minutesPicker'}).props().selectedValue).toBe(
      5,
    );
  });

  it('should render with zeroPadding', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker zeroPadding pickerShows={['hours', 'minutes', 'seconds']} />,
    );

    expect(wrapper.find({testID: 'hoursItem'}).first().props().label).toBe(
      '00 ',
    );
    expect(wrapper.find({testID: 'minutesItem'}).first().props().label).toBe(
      '00 ',
    );
    expect(wrapper.find({testID: 'secondsItem'}).first().props().label).toBe(
      '00 ',
    );
  });

  it('should render pickers with isAmpm', () => {
    const wrapper = shallow<typeof TimePicker>(<TimePicker isAmpm />);
    expect(wrapper.find({testID: 'hoursPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'minutesPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'ampmPicker'}).isEmptyRender()).toBe(false);

    expect(wrapper.find({testID: 'hoursItem'}).length).toBe(12);
    expect(wrapper.find({testID: 'minutesItem'}).length).toBe(60);

    wrapper.find({testID: 'ampmPicker'}).props().onValueChange('pm', 'pm');
    expect(wrapper.find({testID: 'ampmPicker'}).props().selectedValue).toBe(
      'pm',
    );

    expect(wrapper.find({testID: 'amItem'}).props().label).toBe('am');
    expect(wrapper.find({testID: 'pmItem'}).props().label).toBe('pm');
  });

  it('should render pickers with isAmpm & ampmLocalization', () => {
    const wrapper = shallow<typeof TimePicker>(
      <TimePicker isAmpm ampmLocalization={{am: '午前', pm: '午後'}} />,
    );
    expect(wrapper.find({testID: 'hoursPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'minutesPicker'}).isEmptyRender()).toBe(false);
    expect(wrapper.find({testID: 'ampmPicker'}).isEmptyRender()).toBe(false);

    wrapper.find({testID: 'ampmPicker'}).props().onValueChange('pm', '午後');
    expect(wrapper.find({testID: 'ampmPicker'}).props().selectedValue).toBe(
      'pm',
    );

    expect(wrapper.find({testID: 'amItem'}).props().label).toBe('午前');
    expect(wrapper.find({testID: 'pmItem'}).props().label).toBe('午後');
  });
});
