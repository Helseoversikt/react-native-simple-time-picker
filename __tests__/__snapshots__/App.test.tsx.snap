// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactNativeSimpleTimePicker should render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Picker
    mode="dialog"
    onValueChange={[Function]}
    selectedValue={0}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="hoursPicker"
  >
    <PickerItem
      key="0"
      label="0 "
      testID="hoursItem"
      value={0}
    />
    <PickerItem
      key="1"
      label="1 "
      testID="hoursItem"
      value={1}
    />
    <PickerItem
      key="2"
      label="2 "
      testID="hoursItem"
      value={2}
    />
    <PickerItem
      key="3"
      label="3 "
      testID="hoursItem"
      value={3}
    />
    <PickerItem
      key="4"
      label="4 "
      testID="hoursItem"
      value={4}
    />
    <PickerItem
      key="5"
      label="5 "
      testID="hoursItem"
      value={5}
    />
    <PickerItem
      key="6"
      label="6 "
      testID="hoursItem"
      value={6}
    />
    <PickerItem
      key="7"
      label="7 "
      testID="hoursItem"
      value={7}
    />
    <PickerItem
      key="8"
      label="8 "
      testID="hoursItem"
      value={8}
    />
    <PickerItem
      key="9"
      label="9 "
      testID="hoursItem"
      value={9}
    />
    <PickerItem
      key="10"
      label="10 "
      testID="hoursItem"
      value={10}
    />
    <PickerItem
      key="11"
      label="11 "
      testID="hoursItem"
      value={11}
    />
    <PickerItem
      key="12"
      label="12 "
      testID="hoursItem"
      value={12}
    />
    <PickerItem
      key="13"
      label="13 "
      testID="hoursItem"
      value={13}
    />
    <PickerItem
      key="14"
      label="14 "
      testID="hoursItem"
      value={14}
    />
    <PickerItem
      key="15"
      label="15 "
      testID="hoursItem"
      value={15}
    />
    <PickerItem
      key="16"
      label="16 "
      testID="hoursItem"
      value={16}
    />
    <PickerItem
      key="17"
      label="17 "
      testID="hoursItem"
      value={17}
    />
    <PickerItem
      key="18"
      label="18 "
      testID="hoursItem"
      value={18}
    />
    <PickerItem
      key="19"
      label="19 "
      testID="hoursItem"
      value={19}
    />
    <PickerItem
      key="20"
      label="20 "
      testID="hoursItem"
      value={20}
    />
    <PickerItem
      key="21"
      label="21 "
      testID="hoursItem"
      value={21}
    />
    <PickerItem
      key="22"
      label="22 "
      testID="hoursItem"
      value={22}
    />
    <PickerItem
      key="23"
      label="23 "
      testID="hoursItem"
      value={23}
    />
  </Picker>
  <Picker
    mode="dialog"
    onValueChange={[Function]}
    selectedValue={0}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="minutesPicker"
  >
    <PickerItem
      key="0"
      label="0 "
      testID="minutesItem"
      value={0}
    />
    <PickerItem
      key="1"
      label="1 "
      testID="minutesItem"
      value={1}
    />
    <PickerItem
      key="2"
      label="2 "
      testID="minutesItem"
      value={2}
    />
    <PickerItem
      key="3"
      label="3 "
      testID="minutesItem"
      value={3}
    />
    <PickerItem
      key="4"
      label="4 "
      testID="minutesItem"
      value={4}
    />
    <PickerItem
      key="5"
      label="5 "
      testID="minutesItem"
      value={5}
    />
    <PickerItem
      key="6"
      label="6 "
      testID="minutesItem"
      value={6}
    />
    <PickerItem
      key="7"
      label="7 "
      testID="minutesItem"
      value={7}
    />
    <PickerItem
      key="8"
      label="8 "
      testID="minutesItem"
      value={8}
    />
    <PickerItem
      key="9"
      label="9 "
      testID="minutesItem"
      value={9}
    />
    <PickerItem
      key="10"
      label="10 "
      testID="minutesItem"
      value={10}
    />
    <PickerItem
      key="11"
      label="11 "
      testID="minutesItem"
      value={11}
    />
    <PickerItem
      key="12"
      label="12 "
      testID="minutesItem"
      value={12}
    />
    <PickerItem
      key="13"
      label="13 "
      testID="minutesItem"
      value={13}
    />
    <PickerItem
      key="14"
      label="14 "
      testID="minutesItem"
      value={14}
    />
    <PickerItem
      key="15"
      label="15 "
      testID="minutesItem"
      value={15}
    />
    <PickerItem
      key="16"
      label="16 "
      testID="minutesItem"
      value={16}
    />
    <PickerItem
      key="17"
      label="17 "
      testID="minutesItem"
      value={17}
    />
    <PickerItem
      key="18"
      label="18 "
      testID="minutesItem"
      value={18}
    />
    <PickerItem
      key="19"
      label="19 "
      testID="minutesItem"
      value={19}
    />
    <PickerItem
      key="20"
      label="20 "
      testID="minutesItem"
      value={20}
    />
    <PickerItem
      key="21"
      label="21 "
      testID="minutesItem"
      value={21}
    />
    <PickerItem
      key="22"
      label="22 "
      testID="minutesItem"
      value={22}
    />
    <PickerItem
      key="23"
      label="23 "
      testID="minutesItem"
      value={23}
    />
    <PickerItem
      key="24"
      label="24 "
      testID="minutesItem"
      value={24}
    />
    <PickerItem
      key="25"
      label="25 "
      testID="minutesItem"
      value={25}
    />
    <PickerItem
      key="26"
      label="26 "
      testID="minutesItem"
      value={26}
    />
    <PickerItem
      key="27"
      label="27 "
      testID="minutesItem"
      value={27}
    />
    <PickerItem
      key="28"
      label="28 "
      testID="minutesItem"
      value={28}
    />
    <PickerItem
      key="29"
      label="29 "
      testID="minutesItem"
      value={29}
    />
    <PickerItem
      key="30"
      label="30 "
      testID="minutesItem"
      value={30}
    />
    <PickerItem
      key="31"
      label="31 "
      testID="minutesItem"
      value={31}
    />
    <PickerItem
      key="32"
      label="32 "
      testID="minutesItem"
      value={32}
    />
    <PickerItem
      key="33"
      label="33 "
      testID="minutesItem"
      value={33}
    />
    <PickerItem
      key="34"
      label="34 "
      testID="minutesItem"
      value={34}
    />
    <PickerItem
      key="35"
      label="35 "
      testID="minutesItem"
      value={35}
    />
    <PickerItem
      key="36"
      label="36 "
      testID="minutesItem"
      value={36}
    />
    <PickerItem
      key="37"
      label="37 "
      testID="minutesItem"
      value={37}
    />
    <PickerItem
      key="38"
      label="38 "
      testID="minutesItem"
      value={38}
    />
    <PickerItem
      key="39"
      label="39 "
      testID="minutesItem"
      value={39}
    />
    <PickerItem
      key="40"
      label="40 "
      testID="minutesItem"
      value={40}
    />
    <PickerItem
      key="41"
      label="41 "
      testID="minutesItem"
      value={41}
    />
    <PickerItem
      key="42"
      label="42 "
      testID="minutesItem"
      value={42}
    />
    <PickerItem
      key="43"
      label="43 "
      testID="minutesItem"
      value={43}
    />
    <PickerItem
      key="44"
      label="44 "
      testID="minutesItem"
      value={44}
    />
    <PickerItem
      key="45"
      label="45 "
      testID="minutesItem"
      value={45}
    />
    <PickerItem
      key="46"
      label="46 "
      testID="minutesItem"
      value={46}
    />
    <PickerItem
      key="47"
      label="47 "
      testID="minutesItem"
      value={47}
    />
    <PickerItem
      key="48"
      label="48 "
      testID="minutesItem"
      value={48}
    />
    <PickerItem
      key="49"
      label="49 "
      testID="minutesItem"
      value={49}
    />
    <PickerItem
      key="50"
      label="50 "
      testID="minutesItem"
      value={50}
    />
    <PickerItem
      key="51"
      label="51 "
      testID="minutesItem"
      value={51}
    />
    <PickerItem
      key="52"
      label="52 "
      testID="minutesItem"
      value={52}
    />
    <PickerItem
      key="53"
      label="53 "
      testID="minutesItem"
      value={53}
    />
    <PickerItem
      key="54"
      label="54 "
      testID="minutesItem"
      value={54}
    />
    <PickerItem
      key="55"
      label="55 "
      testID="minutesItem"
      value={55}
    />
    <PickerItem
      key="56"
      label="56 "
      testID="minutesItem"
      value={56}
    />
    <PickerItem
      key="57"
      label="57 "
      testID="minutesItem"
      value={57}
    />
    <PickerItem
      key="58"
      label="58 "
      testID="minutesItem"
      value={58}
    />
    <PickerItem
      key="59"
      label="59 "
      testID="minutesItem"
      value={59}
    />
  </Picker>
  <Picker
    mode="dialog"
    onValueChange={[Function]}
    selectedValue={0}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="secondsPicker"
  >
    <PickerItem
      key="0"
      label="0 "
      testID="secondsItem"
      value={0}
    />
    <PickerItem
      key="1"
      label="1 "
      testID="secondsItem"
      value={1}
    />
    <PickerItem
      key="2"
      label="2 "
      testID="secondsItem"
      value={2}
    />
    <PickerItem
      key="3"
      label="3 "
      testID="secondsItem"
      value={3}
    />
    <PickerItem
      key="4"
      label="4 "
      testID="secondsItem"
      value={4}
    />
    <PickerItem
      key="5"
      label="5 "
      testID="secondsItem"
      value={5}
    />
    <PickerItem
      key="6"
      label="6 "
      testID="secondsItem"
      value={6}
    />
    <PickerItem
      key="7"
      label="7 "
      testID="secondsItem"
      value={7}
    />
    <PickerItem
      key="8"
      label="8 "
      testID="secondsItem"
      value={8}
    />
    <PickerItem
      key="9"
      label="9 "
      testID="secondsItem"
      value={9}
    />
    <PickerItem
      key="10"
      label="10 "
      testID="secondsItem"
      value={10}
    />
    <PickerItem
      key="11"
      label="11 "
      testID="secondsItem"
      value={11}
    />
    <PickerItem
      key="12"
      label="12 "
      testID="secondsItem"
      value={12}
    />
    <PickerItem
      key="13"
      label="13 "
      testID="secondsItem"
      value={13}
    />
    <PickerItem
      key="14"
      label="14 "
      testID="secondsItem"
      value={14}
    />
    <PickerItem
      key="15"
      label="15 "
      testID="secondsItem"
      value={15}
    />
    <PickerItem
      key="16"
      label="16 "
      testID="secondsItem"
      value={16}
    />
    <PickerItem
      key="17"
      label="17 "
      testID="secondsItem"
      value={17}
    />
    <PickerItem
      key="18"
      label="18 "
      testID="secondsItem"
      value={18}
    />
    <PickerItem
      key="19"
      label="19 "
      testID="secondsItem"
      value={19}
    />
    <PickerItem
      key="20"
      label="20 "
      testID="secondsItem"
      value={20}
    />
    <PickerItem
      key="21"
      label="21 "
      testID="secondsItem"
      value={21}
    />
    <PickerItem
      key="22"
      label="22 "
      testID="secondsItem"
      value={22}
    />
    <PickerItem
      key="23"
      label="23 "
      testID="secondsItem"
      value={23}
    />
    <PickerItem
      key="24"
      label="24 "
      testID="secondsItem"
      value={24}
    />
    <PickerItem
      key="25"
      label="25 "
      testID="secondsItem"
      value={25}
    />
    <PickerItem
      key="26"
      label="26 "
      testID="secondsItem"
      value={26}
    />
    <PickerItem
      key="27"
      label="27 "
      testID="secondsItem"
      value={27}
    />
    <PickerItem
      key="28"
      label="28 "
      testID="secondsItem"
      value={28}
    />
    <PickerItem
      key="29"
      label="29 "
      testID="secondsItem"
      value={29}
    />
    <PickerItem
      key="30"
      label="30 "
      testID="secondsItem"
      value={30}
    />
    <PickerItem
      key="31"
      label="31 "
      testID="secondsItem"
      value={31}
    />
    <PickerItem
      key="32"
      label="32 "
      testID="secondsItem"
      value={32}
    />
    <PickerItem
      key="33"
      label="33 "
      testID="secondsItem"
      value={33}
    />
    <PickerItem
      key="34"
      label="34 "
      testID="secondsItem"
      value={34}
    />
    <PickerItem
      key="35"
      label="35 "
      testID="secondsItem"
      value={35}
    />
    <PickerItem
      key="36"
      label="36 "
      testID="secondsItem"
      value={36}
    />
    <PickerItem
      key="37"
      label="37 "
      testID="secondsItem"
      value={37}
    />
    <PickerItem
      key="38"
      label="38 "
      testID="secondsItem"
      value={38}
    />
    <PickerItem
      key="39"
      label="39 "
      testID="secondsItem"
      value={39}
    />
    <PickerItem
      key="40"
      label="40 "
      testID="secondsItem"
      value={40}
    />
    <PickerItem
      key="41"
      label="41 "
      testID="secondsItem"
      value={41}
    />
    <PickerItem
      key="42"
      label="42 "
      testID="secondsItem"
      value={42}
    />
    <PickerItem
      key="43"
      label="43 "
      testID="secondsItem"
      value={43}
    />
    <PickerItem
      key="44"
      label="44 "
      testID="secondsItem"
      value={44}
    />
    <PickerItem
      key="45"
      label="45 "
      testID="secondsItem"
      value={45}
    />
    <PickerItem
      key="46"
      label="46 "
      testID="secondsItem"
      value={46}
    />
    <PickerItem
      key="47"
      label="47 "
      testID="secondsItem"
      value={47}
    />
    <PickerItem
      key="48"
      label="48 "
      testID="secondsItem"
      value={48}
    />
    <PickerItem
      key="49"
      label="49 "
      testID="secondsItem"
      value={49}
    />
    <PickerItem
      key="50"
      label="50 "
      testID="secondsItem"
      value={50}
    />
    <PickerItem
      key="51"
      label="51 "
      testID="secondsItem"
      value={51}
    />
    <PickerItem
      key="52"
      label="52 "
      testID="secondsItem"
      value={52}
    />
    <PickerItem
      key="53"
      label="53 "
      testID="secondsItem"
      value={53}
    />
    <PickerItem
      key="54"
      label="54 "
      testID="secondsItem"
      value={54}
    />
    <PickerItem
      key="55"
      label="55 "
      testID="secondsItem"
      value={55}
    />
    <PickerItem
      key="56"
      label="56 "
      testID="secondsItem"
      value={56}
    />
    <PickerItem
      key="57"
      label="57 "
      testID="secondsItem"
      value={57}
    />
    <PickerItem
      key="58"
      label="58 "
      testID="secondsItem"
      value={58}
    />
    <PickerItem
      key="59"
      label="59 "
      testID="secondsItem"
      value={59}
    />
  </Picker>
</View>
`;
